packages/models-library/tests/test_utils_models_factory.py

```diff
@@ -0,0 +1,303 @@
+
+    def _create(field_name) -> classmethod:
+        return validator(field_name, allow_reuse=True)(_map[field_name])
+
+    return _create
+
+
+create_validator_for = _validators_factory()
+
+#
+# NOTE: Rationale of this test-suite
+#
+# Below we represent different views of a 'user' resource represented with
+# different models depending on the context. We have a domain model 'User', that
+# is used to exchange internally in the business logic, as well as different
+# views used in the request body (e.g. 'UserCreate', 'UserUpdate', ...) or response payload
+# (e.g. 'UserGet', 'UserListItem', ...) for CRUD entrypoints in an API.
+#
+# Note that every context demands not only a different set of fields but also
+# different constraints. Those will depend on nature of the parsed data sources
+# as well as the guarantees defined on the data captured in the model.
+#
+# This approach should be applicable to any resource but we find that
+# 'user' is a good use case that naturally incorporates many of the variants
+# that we have typically encountered.
+#
+# All these variants have many features in common so the idea is to implement a minimalistic
+# policy-based tools that can safely compose them all.
+#
+# Good examples on how to use model polices can be found
+#  in https://fastapi-crudrouter.awtkns.com/schemas or
+#  in https://fastapi.tiangolo.com/tutorial/body-updates/#body-updates
+#
+#
+class User(BaseModel):
+    """Domain model"""
+
+    id: PositiveInt
+    display_name: str
+    username: str
+    password_hash: str
+
+    # validators when model created in code
+    _name_must_contain_space = create_validator_for("display_name")
+    _username_alphanumeric = create_validator_for("username")
+
+
+class UserCreate(BaseModel):
+    """in -> Model for body of POST /users"""
+
+    display_name: str
+    username: str
+    password: str
+    password2: str
+
+    # parses json-body from Create request
+    _name_must_contain_space = create_validator_for("display_name")
+    _username_alphanumeric = create_validator_for("username")
+    _passwords_match = create_validator_for("password2")
+
+
+class UserUpdate(BaseModel):
+    """in -> Model for body of PATCH /users/{id}"""
+
+    display_name: Optional[str]
+    username: Optional[str]
+    password: Optional[str]
+    password2: Optional[str]
+
+    # parses json-body from Update request
+    _name_must_contain_space = create_validator_for("display_name")
+    _username_alphanumeric = create_validator_for("username")
+    _passwords_match = create_validator_for("password2")
+
+
+# Model for body of PUT /users/{id}
+UserReplace = UserCreate
+
+
+class UserGet(BaseModel):
+    """<- out Detailed model for response in GET /users/{id}"""
+
+    id: PositiveInt
+    display_name: str
+    username: str
+
+    # parses from User (i.e. validated domain model)
+
+
+class UserListItem(BaseModel):
+    """<- out Item model for response in GET /users"""
+
+    id: PositiveInt
+    username: str
```

Comment:
not sure I understand here how or why you remove display_name. but this is just an example, it's flexible right?

Reply:
it is an example. I just wanted to create a "lighter"/"non-detailed" version of ``UserGet`` for the lists to demostrate the flexibility